receiver/redisreceiver/redis_runnable.go

```diff
@@ -104,7 +107,7 @@ func (r *redisRunnable) Run() error {
 		)
 	}
 
-	md := newMetricsData(metrics)
+	md := newMetricsData(metrics, r.serverName)
 
 	err = r.metricsConsumer.ConsumeMetricsData(r.ctx, *md)
 	numTimeSeries, numPoints := obsreport.CountMetricPoints(*md)
```

Comment:
You can simply defer obsreport.EndMetricsReceiveOp(ctx, dataformat, numPoints, numTimeSeries, err) instead of adding it every time you return.

Reply:
I punted on this one for now. Hoping to discuss with you tomorrow.